misk-hibernate/src/main/kotlin/misk/hibernate/RealTransacter.kt

```diff
@@ -102,6 +111,8 @@ internal class RealTransacter private constructor(
       try {
         val result = lambda(session)
 
+        // Flush any changes to the databased before commit
+        session.hibernateSession.flush()
```

Comment:
hmm I can't quite speak to the impact of adding a flush here so I'd want some additional eyes on this point. Why is this needed? Doesn't the transaction commit take care of that for you.

Also: databased -> database on the comment

Reply:
my guess is so the query listeners fire, although I would expect them to fire when committing